src/main.rs

```diff
@@ -17,16 +17,16 @@ struct Args {
     pub end_date: Option<String>,
 
     #[clap()]
-    /// numerical value for a region (1-17) or first part of a UK postcode
-    pub value: String,
+    /// numerical value for a region (1-17), first part of a UK postcode
+    /// or national data if left empty
+    pub value: Option<String>,
```

Comment:
I think we should avoid `Option` unless is necessary and default to `National` as unwrapping and what not can easily get unwieldily.

I've started to do some tweaks toward that direction [here](https://github.com/jnioche/carbonintensity-api/pull/26) but broadly speaking:
1. `value` should be of type `Target`
  - I've made this change in that PR
  - I've renamed to `target` there as I think it's what this argument is about ☺️
2. we should use `#[arg(default_value_t=Target::National)` to default to national when not provided
3. in order for that to work Target needs to implement `From<String>` instead of `FromStr` which would do almost the same but it is infallible...but in our case we can do that as we can default to national when `s.trim().is_empty()`
4. you can then just use `args.target` without the need for further processing/checking/unwrapping ☺️ 

and Clap/Rust's traits would just work their magic...happy to clarify if it doesn't make quite sense :)

Reply:
it does make sense, thanks. What about merging in #26 first then amend this PR accordingly?